tests/geth/test_running_geth_node.py

```diff
@@ -67,14 +67,14 @@ def test_running_node_and_mining(project_test04, open_port):
     data_dir = get_geth_data_dir(project_test04, 'default')
 
     command, proc = run_geth_node(data_dir, rpc_port=open_port, mine=True)
-    time.sleep(2)
+    wait_for_popen(proc)
     rpc_client = Client('127.0.0.1', port=open_port)
     block_num = rpc_client.get_block_number()
     start = time.time()
-    while time.time() < start + 10:
+    while rpc_client.get_block_number() <= block_num:
```

Comment:
This can actually be done with `rpc_client.wait_for_block(block_num)` which will block for up to 60 seconds waiting for the block number.